client/app/bundles/course/assessment/submission/components/answers/index.tsx

```diff
@@ -0,0 +1,161 @@
+// eslint-disable-next-line simple-import-sort/imports
+import { memo } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { Divider, Typography } from '@mui/material';
+import equal from 'fast-deep-equal';
+import { FIELD_LONG_DEBOUNCE_DELAY_MS } from 'lib/constants/sharedConstants';
+import { useAppDispatch } from 'lib/hooks/store';
+import { useDebounce } from 'lib/hooks/useDebounce';
+import { SubmissionQuestionData } from 'types/course/assessment/submission/question/types';
+
+import { QuestionType } from 'types/course/assessment/question';
+import { saveAnswer, updateClientVersion } from '../../actions/answers';
+import { uploadTextResponseFiles } from '../../actions/answers/textResponse';
+
+import Answer from './Answer';
+import AnswerHeader from './AnswerHeader';
+import { AnswerPropsMap } from './types';
+import { QuestionHistory } from '../../reducers/history/types';
+
+interface SubmissionAnswerProps<T extends keyof typeof QuestionType> {
+  answerId: number;
+  graderView: boolean;
+  historyQuestions: Record<number, QuestionHistory>;
+  question: SubmissionQuestionData<T>;
+  questionType: T;
+  readOnly: boolean;
+  showMcqMrqSolution: boolean;
+}
+
+const SubmissionAnswer = <T extends keyof typeof QuestionType>(
+  props: SubmissionAnswerProps<T>,
+): JSX.Element => {
+  const {
+    answerId,
+    graderView,
+    historyQuestions,
+    question,
+    questionType,
+    readOnly,
+    showMcqMrqSolution,
+  } = props;
+  const dispatch = useAppDispatch();
+
+  const { getValues, resetField } = useFormContext();
+
+  const handleSaveAnswer = (
+    answerData: unknown,
+    savedAnswerId: number,
+    currentTime: number,
+  ): void => {
+    dispatch(saveAnswer(answerData, savedAnswerId, currentTime, resetField));
+  };
+
+  const debouncedSaveAnswer = useDebounce(
+    handleSaveAnswer,
+    questionType === 'VoiceResponse' ? 0 : FIELD_LONG_DEBOUNCE_DELAY_MS,
+    [],
+  );
+
+  const saveAnswerAndUpdateClientVersion = (saveAnswerId: number): void => {
+    const answer = getValues()[saveAnswerId];
+    const currentTime = Date.now();
+    dispatch(updateClientVersion(saveAnswerId, currentTime));
```

Comment:
I personally tried the interaction on the web. It feels weird that the saving is very long -> psychologically it prevents me to edit the content further. So i only start the saving indicator when it’s actually saving.

Reply:
I understand your point, but actually even with the saving indicator being activated after it actually saves the answer, the user can still do the changing and there should be no error in doing that (since we have handled the race condition that might happen here). Also, the reason we decided in having such requirements is to inform user that we have the autosaving features taking place.

Iff the problem lies in saving feels taking quite a long time, then maybe we can consider having the smaller number for debounce delay time?

Reply:
>> the user can still do the changing and there should be no error 
I am not saying they cant change, I was saying i would feel less inclined to change as a user.

>> inform user that we have the autosaving features taking place.
This reason is not valid as the loading will take place anyway and they will know. If the loading indicator appears a bit later, does that mean there’s no autosaving?

>> smaller number debounce time
What’s the purpose of having debounce here? It’s not for UIUX purpose, but to ensure we are not spamming our backend too often.

Reply:
understand on the issue here. as per discussed this afternoon, basically while we refer to the behavior of the savingStatus in Google Form for this, it's quite different since the saving indicator appears on every answer in contrast of what Google Form has, which is on top of the form

because of this difference, you have the point in which it creates the negative psychological impact for the users when they want to update the answer and immediately the Saving indicator appears.

With that in say, I think it's safe to let the behavior as it is (to render the saving indicator when it's actually saving)